README.md

```diff
@@ -21,7 +21,7 @@ Other useful resources:
 Table of contents
 =================
 
-  * [Try out Mattermost](#try-out-mattermost)
+  * [Try out Mattermost](https://mattermost.com/sign-up/)
```

Comment:
This particular section title is not required anymore and can be deleted instead. It was used way back when we had a section https://github.com/mattermost/mattermost-server/tree/d8986840aca26f9e32d67ba5db9fcbf959d83338#try-out-mattermost